sdk/storage/Azure.Storage.Common/src/Shared/StorageSharedKeyCredentialInternals.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage
+{
+    /// <summary>
+    /// This class is added to access protected static methods off of the base class
+    /// that should not be exposed directly to customers.
+    /// </summary>
+    internal class StorageSharedKeyCredentialInternals : StorageSharedKeyCredential
+    {
+        #pragma warning disable IDE0051 // Remove unused private members
```

Comment:
Can't you make this `private StorageSharedKeyCredentialInternals() { }` and skip the pragma?

Reply:
Not without still passing the required params to the base constructor.